src/AllowList.php

```diff
@@ -5,92 +5,48 @@
 namespace Laminas\Filter;
 
 use Laminas\Stdlib\ArrayUtils;
-use Traversable;
 
+use function array_values;
 use function in_array;
-use function is_array;
 
 /**
  * @psalm-type Options = array{
  *     strict?: bool,
- *     list?: array,
- *     ...
+ *     list?: iterable<array-key, mixed>,
  * }
- * @extends AbstractFilter<Options>
+ * @implements FilterInterface<null>
  */
-final class AllowList extends AbstractFilter
+final class AllowList implements FilterInterface
 {
-    /** @var bool */
-    protected $strict = false;
+    private readonly bool $strict;
+    /** @var list<mixed> */
+    private readonly array $list;
 
-    /** @var array */
-    protected $list = [];
-
-    /**
-     * @param null|array|Traversable $options
-     */
-    public function __construct($options = null)
-    {
-        if (null !== $options) {
-            $this->setOptions($options);
-        }
-    }
-
-    /**
-     * Determine whether the in_array() call should be "strict" or not. See in_array docs.
-     *
-     * @param bool $strict
-     */
-    public function setStrict($strict = true): void
+    /** @param Options $options */
+    public function __construct(array $options = [])
     {
-        $this->strict = (bool) $strict;
+        $this->strict = $options['strict'] ?? false;
+        $list         = ArrayUtils::iteratorToArray($options['list'] ?? []);
```

Comment:
Does this filter even make sense, if the list is not provided? Would it make sense to make the option mandatory?

Reply:
You're right - the filter doesn't make sense without a mandatory list 👍

Reply:
But… Options need to be optional because every filter needs to be newable without arguments